tensorboard/plugins/base_plugin.py

```diff
@@ -97,6 +97,16 @@ def frontend_metadata(self):
         """
         return FrontendMetadata()
 
+    def relevant_summary_plugins(self):
```

Comment:
Design question: would it be worth exposing this instead on TBLoader?

It seems reasonable to assume that in general we expect this to be a fixed list / static property of the plugin, rather than something dynamically determined.  This would allow us to defer loading of the plugin until we encounter an experiment that actually needs it, which could help the backend start up faster and use less memory in a world where there are many dynamic plugins and they're more heavyweight.  We probably don't get much advantage from it right now (where the summary-based plugins tend to be the lightweight ones anyway), and there are some downsides as well (if the plugin takes a while to load or has an import-time error, this impact occurs later on, which might be bad), but I wonder if it might be more future-proof that way.

WDYT?

Reply:
Yeah, I agree it makes sense to defer worrying about this for the points you made.  I suspect if we really wanted to do proper dynamic plugin loading like that we'd maybe want to revisit the TBLoader/TBPlugin breakdown more holistically anyway.  But useful to hear your thoughts about this, thanks.